mlibrary/search
# Update packages (March) - Part 3

# Overview
This pull request removes several unused dependencies and updates the `README.md` file.

## Removed Dependencies

### `deepcopy`
The `deepcopy` function has been replaced with the `_.clone` function from the already existing `underscore` dependency. `_.clone` produces a shallow copy, but with how `deepcopy` was being used, it produced identical results.

### `@babel/polyfill`
From https://babeljs.io/docs/en/babel-polyfill:
> 🚨 As of Babel 7.4.0, this package has been deprecated in favor of directly including `core-js/stable` (to polyfill ECMAScript features) and `regenerator-runtime/runtime` (needed to use transpiled generator functions):

`core-js` already exists as a dependency. `regenerator-runtime` has been added for this replacement.

### `@reach/visually-hidden`
This created a component that produced a `<span>` tag with CSS commonly used to accessibly hide elements. The dependency has been removed, and the components have been replaced with HTML and CSS.

### `react-responsive`
This created a component that watched for `window` resizing, and would add/remove elements from the `DOM` based on a certain size. This was only used in `src/modules/pages/components/DatastorePage/container.js`. I replaced the dependency by restructuring the file using just HTML and CSS to hide/show.

### `remark` and `remark-react`
These dependencies converted markup to HTML, and was only used on the [Technical Overview](http://localhost:3000/technical-overview) page. The page has been rebuilt, no longer using markup and is modeled after how [Accessibility](http://localhost:3000/accessibility) was built.

## Future Removal

### `react-accessible-accordion`
`react-accessible-accordion` can be replaced with using the `<details>` HTML element. It is currently being used in two locations:
- `src/modules/filters/components/Filters/index.js`
- `src/modules/resource-acccess/components/holders.js`

This would be about a days worth of work.

### `react-autosuggest`
This is only being used in `src/modules/favorites/components/FavoriteInputTag/index.js`, which is currently in the process of being retired. Once that is complete, this dependency can be removed.

## Testing
- `npm run reinstall && npm start`
- Poke around [the site](http://localhost:3000/) and see if anything seems broken.
- Check the [Technical Overview](http://localhost:3000/technical-overview) page and see if it looks fine.
- Search for an item and use the filters. Check on large and small screens to see if the filtering still works fine.
